sapi/fpm/fpm/fpm_sockets.c

```diff
@@ -90,7 +90,10 @@ static void fpm_sockets_cleanup(int which, void *arg) /* {{{ */
 				}
 			}
 
-			p += sprintf(env_value + p + socket_set_buf, "%s%s=%s", (p && !socket_set_buf) ? "," : "", ls->key, fd);
+			int written = snprintf(env_value + p + socket_set_buf, 
```

Comment:
Similarly, the smart_string API would avoid manually reallocating a buffer.

Reply:
It is. You just included a header that no longer exists with PHP 8.5 and you mixed the smart_string API (char*) with the smart_str API (zend_string*).

Reply:
I was deliberate when I said “smart_string” in my initial comment. Given that you are not actually interested in a zend_string, the smart_string API is better suited than the smart_str API for this use case.

Reply:
Ah, I wasn't aware that there were two different APIs. PR updated